test/conftest.py

```diff
@@ -11,33 +9,32 @@
 @pytest.fixture(scope="session")
 def db_connection():
     """
-    Establishes odbc connection for the whole session
-    :return: pyodbc connection class
+    Establishes postgresql connection for the whole session
+    :return: Connection_Manager object
     """
-    env_path = "db/.env"
+    env_path = ".env"
     if os.path.exists(env_path):
+        print("here")
         env_values = dotenv_values(env_path)
-        env_variables = ["POSTGRES_USER", "POSTGRES_DB", "POSTGRES_PASSWORD"]
+        env_variables = [
+            "TEST_POSTGRES_USER",
+            "TEST_POSTGRES_DB",
+            "TEST_POSTGRES_PASSWORD",
+            "TEST_POSTGRES_PORT",
+            "TEST_POSTGRES_HOST",
+        ]
         for env_variable in env_variables:
             os.environ[env_variable] = env_values.get(env_variable)
 
-    db_connection_config = {
-        "host": "localhost",
-        "dbname": os.getenv("POSTGRES_DB"),
-        "user": os.getenv("POSTGRES_USER"),
-        "password": os.getenv("POSTGRES_PASSWORD"),
-    }
     testdb_connection_config = {
-        "host": "localhost",
-        "dbname": "testdb",
-        "user": os.getenv("POSTGRES_USER"),
-        "password": os.getenv("POSTGRES_PASSWORD"),
+        "host": os.getenv("TEST_POSTGRES_HOST"),
+        "dbname": os.getenv("TEST_POSTGRES_DB"),
+        "user": os.getenv("TEST_POSTGRES_USER"),
+        "password": os.getenv("TEST_POSTGRES_PASSWORD"),
+        "port": os.getenv("TEST_POSTGRES_PORT"),
```

Comment:
My understanding of this code is that you always read from file `.env`, write the variables from the file into the environment and then you get the same variables from the environment again for your testdb connection string.

If that's correct, do you think we could do it a bit differently? E.g.
- if the test service is spinned up, then your docker compose has added environment variables "TEST_POSTGRES_DB" etc. already to the environment. Maybe we could try to read that first and then if it fails, check the `.env` file.
- if we're reading the `.env` file, maybe we can just read the variable from there and just give that to the connection string instead. Would that work? From a quick check, I don't see anywhere else you need those variables in the environment, but just to establish the connection and you do that with the config dictionary.

Reply:
The thought here was that the .env file is not copied into any container (you can see that I only copy the subfolders in Dockerfile, not the main folder of the project). When using docker-compose, the .env file is only read by compose.yaml during build, the environment variables are read and passed to the different containers.
The only exception is if we want to run the code locally, i.e. not in a container. In that case the if-statement in line 16 will not fail, and the environment variables are instead read from the local .env file.